letta/client/client.py

```diff
@@ -2288,16 +2253,14 @@ def update_tool(
         Returns:
             tool (Tool): Updated tool
         """
-        if func:
-            source_code = parse_source_code(func)
-        else:
-            source_code = None
-
-        source_type = "python"
+        update_data = {
```

Comment:
why not just pass this into `ToolUpdate`?

Reply:
sorry this code is very hard to understand, let me refactor it - basically trying to only include keys that are non-null